payments/models.py

```diff
@@ -0,0 +1,139 @@
+DEFAULT_TAX_PERCENTAGE = Decimal("24.0")
+
+
+class AbstractBaseProduct(TimeStampedModel, UUIDModel):
+    price_value = models.DecimalField(
+        verbose_name=_("price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+    )
+    price_unit = EnumField(PriceUnits, default=PriceUnits.AMOUNT)
+    tax_percentage = models.DecimalField(
+        verbose_name=_("tax percentage"),
+        max_digits=5,
+        decimal_places=2,
+        default=DEFAULT_TAX_PERCENTAGE,
+        choices=[(tax, str(tax)) for tax in TAX_PERCENTAGES],
+    )
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+
+class BerthPriceGroup(UUIDModel):
+    name = models.CharField(verbose_name=_("berth price group name"), max_length=128)
+
+    def __str__(self):
+        return f"{self.name} ({self.id})"
+
+
+class AbstractPlaceProduct(AbstractBaseProduct):
+    price_unit = EnumField(
+        PriceUnits,
+        choices=[(PriceUnits.AMOUNT, PriceUnits.AMOUNT.label)],
+        default=PriceUnits.AMOUNT,
+    )
+    tax_percentage = models.DecimalField(
+        verbose_name=_("tax percentage"),
+        max_digits=5,
+        decimal_places=2,
+        default=DEFAULT_TAX_PERCENTAGE,
+        choices=[(DEFAULT_TAX_PERCENTAGE, str(DEFAULT_TAX_PERCENTAGE))],
+    )
+
+    class Meta:
+        abstract = True
+
+
+class BerthProduct(AbstractPlaceProduct):
+    price_group = models.ForeignKey(
+        BerthPriceGroup,
+        on_delete=models.CASCADE,
+        verbose_name=_("price group"),
+        related_name="products",
+    )
+    harbor = models.ForeignKey(
+        "resources.Harbor",
+        verbose_name=_("harbor"),
+        related_name="product",
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["price_group", "harbor"],
+                name="unique_product_for_harbor_pricegroup",
+            ),
+        ]
+
+    def __str__(self):
+        return f"{self.price_group} - {self.harbor} ({self.price_value}€)"
+
+
+class WinterStorageProduct(AbstractPlaceProduct):
+    winter_storage_area = models.OneToOneField(
+        "resources.WinterStorageArea",
+        verbose_name=_("winter storage area"),
+        related_name="product",
+        on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        return f"{self.winter_storage_area} ({self.price_value}€)"
+
+
+class AdditionalProduct(AbstractBaseProduct):
+    service = EnumField(ServiceType, verbose_name=_("service"), max_length=40)
+    period = EnumField(PeriodType, max_length=8)
+
+    @property
+    def product_type(self):
+        if self.service.is_fixed_service():
+            return AdditionalProductType.FIXED_SERVICE
+        elif self.service.is_optional_service():
+            return AdditionalProductType.OPTIONAL_SERVICE
+        return None
```

Comment:
hmhmhm, not a fan of returning `None`, I would rather do
```python
if self.service.is_fixed_service():
    return AdditionalProductType.FIXED_SERVICE
else:
    return AdditionalProductType.OPTIONAL_SERVICE
```
'cause at. the API we. would need to have a filter for this and would be nice in `product_type` could be only either ... or ... .

Reply:
Yes, I agree, and if everything is done right (which it should), it _won't ever_ fall on the `None` case, except when creating a new instance from the Admin.

If you have `else OPTIONAL_SERVICE`, the admin will display `Product type: Optional Service` when creating a new instance, which can be missleading. 

<img width="573" alt="image" src="https://user-images.githubusercontent.com/15201480/82032925-a8433080-96a4-11ea-8aab-755149d67204.png">

Reply:
aahhh, okay, yeah, let's keep it that way